packages/testkit-backend/src/request-handlers.js

```diff
@@ -174,64 +176,67 @@ export function SessionRun (context, data, wire) {
   wire.writeResponse(responses.Result({ id }))
 }
 
-export function ResultNext (context, data, wire) {
+export function ResultNext (neo4j, context, data, wire) {
   const { resultId } = data
   const result = context.getResult(resultId)
+  const binder = new CypherNativeBinders(neo4j)
   if (!('recordIt' in result)) {
     result.recordIt = result[Symbol.asyncIterator]()
   }
   return result.recordIt.next().then(({ value, done }) => {
     if (done) {
       wire.writeResponse(responses.NullRecord())
     } else {
-      wire.writeResponse(responses.Record({ record: value }))
+      wire.writeResponse(responses.Record({ record: value }, { binder }))
     }
   }).catch(e => {
     console.log('got some err: ' + JSON.stringify(e))
     wire.writeError(e)
   })
 }
 
-export function ResultPeek (context, data, wire) {
+export function ResultPeek (neo4j, context, data, wire) {
   const { resultId } = data
   const result = context.getResult(resultId)
+  const binder = new CypherNativeBinders(neo4j)
   if (!('recordIt' in result)) {
     result.recordIt = result[Symbol.asyncIterator]()
   }
   return result.recordIt.peek().then(({ value, done }) => {
     if (done) {
       wire.writeResponse(responses.NullRecord())
     } else {
-      wire.writeResponse(responses.Record({ record: value }))
+      wire.writeResponse(responses.Record({ record: value }, { binder }))
     }
   }).catch(e => {
     console.log('got some err: ' + JSON.stringify(e))
     wire.writeError(e)
   })
 }
 
-export function ResultConsume (context, data, wire) {
+export function ResultConsume (neo4j, context, data, wire) {
   const { resultId } = data
   const result = context.getResult(resultId)
+  const binder = new CypherNativeBinders(neo4j)
 
   return result.summary().then(summary => {
-    wire.writeResponse(responses.Summary({ summary }))
+    wire.writeResponse(responses.Summary({ summary }, { binder }))
   }).catch(e => wire.writeError(e))
 }
 
-export function ResultList (context, data, wire) {
+export function ResultList (neo4j, context, data, wire) {
   const { resultId } = data
-
+  const binder = new CypherNativeBinders(neo4j)
```

Comment:
Why are we instantiating a CypherNativeBinders object in nearly every call? Couldn't this a singleton or a class with a bunch of static methods or such (\<insert JS idiomatic thing\>)? It's stateless, from what I've seen.

Reply:
I could make it part of the context.